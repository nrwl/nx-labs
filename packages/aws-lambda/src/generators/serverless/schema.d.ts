import { Linter } from '@nrwl/linter';
export interface Schema {
  name: string;
  skipFormat?: boolean;
  skipPackageJson?: boolean;
  directory?: string;
  unitTestRunner?: 'jest' | 'none';
  linter?: Linter;
  tags?: string;
  lintTarget?: string;
  buildTarget?: string;
  deployTarget?: string;
}
