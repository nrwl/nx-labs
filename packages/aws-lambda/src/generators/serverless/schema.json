{
  "$schema": "http://json-schema.org/schema",
  "$id": "SchematicsNxServerlessApp",
  "title": "Nx AWS Lambda Serverless Application Options Schema",
  "description": "Nx AWS Lambda Serverless Application Options Schema.",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the node application?",
      "x-priority": "important"
    },
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "x-priority": "important"
    },
    "skipFormat": {
      "description": "Skip formatting files",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`.",
      "x-priority": "internal"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "jest"
    },
    "deployTarget": {
      "description": "The name of the deploy target",
      "type": "string",
      "default": "deploy"
    },
    "lintTarget": {
      "description": "The name of the lint target",
      "type": "string",
      "default": "lint"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the application (used for linting)."
    }
  },
  "required": ["name"]
}
