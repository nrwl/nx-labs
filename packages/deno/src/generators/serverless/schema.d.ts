export interface ServerlessGeneratorSchema {
  name: string;
  tags?: string;
  directory?: string;
  platform?: 'netlify' | 'deno-deploy' | 'none';
  linter?: 'deno' | 'none';
  unitTestRunner?: 'deno' | 'none';
  site?: string;
}
