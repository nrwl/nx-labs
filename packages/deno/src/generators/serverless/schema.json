{
  "$schema": "http://json-schema.org/schema",
  "$id": "DenoServerlessGenerator",
  "title": "Create a Deno Serverless Application",
  "description": "Create an app that is setup to run in a serverless environment.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the application.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?",
      "pattern": "^[a-zA-Z].*$"
    },
    "unitTestRunner": {
      "type": "string",
      "description": "Add a testing target for this application.",
      "enum": ["deno", "none"],
      "default": "deno"
    },
    "linter": {
      "type": "string",
      "description": "Add a linting target for this application.",
      "enum": ["deno", "none"],
      "default": "deno"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d",
      "x-priority": "important"
    },
    "platform": {
      "type": "string",
      "description": "Setup serverless support for a specific platform.",
      "enum": ["netlify", "none"],
      "x-prompt": "Which platform would you like to use?",
      "default": "netlify"
    },
    "site": {
      "description": "The site/project name to deploy to",
      "type": "string",
      "x-priority": "important"
    }
  },
  "required": ["name"]
}
