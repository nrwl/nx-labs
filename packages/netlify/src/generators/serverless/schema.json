{
  "$schema": "http://json-schema.org/schema",
  "$id": "SchematicsNxServerlessApp",
  "title": "Nx Netlify Serverless Application Options Schema",
  "description": "Nx Netlify Serverless Application Options Schema.",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the node application?",
      "x-priority": "important"
    },
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "x-priority": "important"
    },
    "skipFormat": {
      "description": "Skip formatting files",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`.",
      "x-priority": "internal"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "jest"
    },
    "buildTarget": {
      "description": "The name of the build target",
      "type": "string",
      "default": "build"
    },
    "deployTarget": {
      "description": "The name of the deploy target",
      "type": "string",
      "default": "deploy"
    },
    "lintTarget": {
      "description": "The name of the lint target",
      "type": "string",
      "default": "lint"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the application (used for linting)."
    },
    "site": {
      "description": "The site where the serverless function will be deployed",
      "type": "string"
    }
  },
  "required": ["name"]
}
