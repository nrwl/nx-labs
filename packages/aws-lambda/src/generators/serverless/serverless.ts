import {
  convertNxGenerator,
  extractLayoutDirectory,
  GeneratorCallback,
  names,
  runTasksInSerial,
  Tree,
} from '@nx/devkit';
import { Linter } from '@nx/linter';
import { applicationGenerator } from '@nx/node';
import { setupServerlessGenerator } from '../setup-serverless/setup-serverless';
import { Schema } from './schema';

function normalizeOptions(options: Schema): Schema {
  const { projectDirectory } = extractLayoutDirectory(options.directory);
  const appDirectory = projectDirectory
    ? `${names(projectDirectory).fileName}/${names(options.name).fileName}`
    : names(options.name).fileName;

  const appProjectName = appDirectory.replace(new RegExp('/', 'g'), '-');
  return {
    ...options,
    name: names(appProjectName).fileName,
    lintTarget: options.lintTarget ?? 'lint',
    linter: options.linter ?? Linter.EsLint,
    unitTestRunner: options.unitTestRunner ?? 'jest',
  };
}

export async function serverlessGenerator(tree: Tree, schema: Schema) {
  const options = normalizeOptions(schema);
  const tasks: GeneratorCallback[] = [];

  const init = await applicationGenerator(tree, {
    ...options,
    framework: 'none',
    e2eTestRunner: 'none',
    rootProject: true,
  });

  tasks.push(init);

  const addAwsLamdaTask = await setupServerlessGenerator(tree, options);

  tasks.push(addAwsLamdaTask);

  return runTasksInSerial(...tasks);
}

export default serverlessGenerator;
export const serverlessGeneratorSchematic =
  convertNxGenerator(serverlessGenerator);
