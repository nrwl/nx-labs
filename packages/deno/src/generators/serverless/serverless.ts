import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  ProjectConfiguration,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
  writeJson,
} from '@nx/devkit';
import { join, relative } from 'path';
import initDeno from '../init/generator';
import denoSetupFunctions from '../setup-functions/setup-functions';
import { ServerlessGeneratorSchema } from './schema';

type NormalizedSchema = ReturnType<typeof normalizeOptions>;

export async function denoServerlessGenerator(
  tree: Tree,
  options: ServerlessGeneratorSchema
) {
  const opts = normalizeOptions(tree, options);

  await initDeno(tree);
  addProject(tree, opts);
  addFiles(tree, opts);
  const setupTask = await setupDeployConfig(tree, opts);

  await formatFiles(tree);

  return setupTask;
}

function normalizeOptions(tree: Tree, options: ServerlessGeneratorSchema) {
  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const projectRoot = `${getWorkspaceLayout(tree).appsDir}/${projectDirectory}`;
  const rootProject = projectRoot === '.' || projectRoot === '';
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  options.platform ??= 'none';
  options.linter ??= 'deno';
  options.unitTestRunner ??= 'deno';

  return {
    ...options,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
    rootProject,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  const projectConfig = readProjectConfiguration(tree, options.projectName);
  const offset = offsetFromRoot(options.projectRoot);
  const templateOptions = {
    ...options,
    ...names(options.name),
    srcDir: relative(projectConfig.root, projectConfig.sourceRoot),
    tmpl: '',
    offset,
  };
  generateFiles(
    tree,
    join(__dirname, 'files', `platform-${options.platform}`),
    options.projectRoot,
    templateOptions
  );

  writeJson(tree, joinPathFragments(options.projectRoot, 'deno.json'), {
    importMap: `${offset}import_map.json`,
  });
}

export function addProject(tree: Tree, opts: NormalizedSchema) {
  const coverageDirectory = joinPathFragments(
    'coverage',
    opts.rootProject ? opts.name : opts.projectRoot
  );

  const targets: ProjectConfiguration['targets'] = {
    build: {
      executor: '@nx/deno:emit',
      outputs: [
        joinPathFragments(
          'dist',
          opts.rootProject ? opts.name : opts.projectRoot
        ),
      ],
      options: {
        main: joinPathFragments(opts.projectRoot, 'src/main.ts'),
        outputFile: joinPathFragments(
          'dist',
          opts.rootProject ? opts.name : opts.projectRoot,
          'main.js'
        ),
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
    serve: {
      executor: '@nx/deno:run',
      options: {
        buildTarget: `${opts.projectName}:build`,
      },
    },
    test: {
      executor: '@nx/deno:test',
      outputs: [coverageDirectory],
      options: {
        coverageDirectory,
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
    lint: {
      executor: '@nx/deno:lint',
      options: {
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
  };

  if (opts.platform === 'netlify') {
    // No reason to build on netlify as the project sourceRoot is the direct script ran
    delete targets.build;
  }

  if (opts.linter === 'none') {
    delete targets.lint;
  }

  if (opts.unitTestRunner === 'none') {
    delete targets.test;
  }

  addProjectConfiguration(tree, opts.projectName, {
    root: opts.projectRoot,
    projectType: 'application',
    name: opts.projectName,
    sourceRoot: joinPathFragments(opts.projectRoot, 'src'),
    targets,
    tags: opts.parsedTags,
  });
}

async function setupDeployConfig(tree: Tree, opts: NormalizedSchema) {
  const setupTask = await denoSetupFunctions(tree, {
    project: opts.projectName,
    platform: opts.platform,
  });
  // delete fn folder from setup-functions
  tree.delete(joinPathFragments(opts.projectRoot, 'functions'));

  const projectConfig = readProjectConfiguration(tree, opts.projectName);

  if (projectConfig.targets?.['serve-functions']) {
    projectConfig.targets.serve = projectConfig.targets['serve-functions'];
    delete projectConfig.targets['serve-functions'];
  }

  updateProjectConfiguration(tree, opts.projectName, projectConfig);

  return setupTask;
}

export default denoServerlessGenerator;
