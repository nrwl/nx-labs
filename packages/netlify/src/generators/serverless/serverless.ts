import {
  extractLayoutDirectory,
  GeneratorCallback,
  names,
  Tree,
} from '@nrwl/devkit';
import { Linter } from '@nrwl/linter';
import { applicationGenerator } from '@nrwl/node';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import setupServerlessGenerator from '../setup-serverless/setup-serverless';
import { Schema } from './schema';

function normalizeOptions(options: Schema): Schema {
  const { projectDirectory } = extractLayoutDirectory(options.directory);
  const appDirectory = projectDirectory
    ? `${names(projectDirectory).fileName}/${names(options.name).fileName}`
    : names(options.name).fileName;

  const appProjectName = appDirectory.replace(new RegExp('/', 'g'), '-');
  return {
    ...options,
    name: names(appProjectName).fileName,
    lintTarget: options.lintTarget ?? 'lint',
    linter: options.linter ?? Linter.EsLint,
    unitTestRunner: options.unitTestRunner ?? 'jest',
  };
}

export async function serverlessGenerator(tree: Tree, schema: Schema) {
  const options = normalizeOptions(schema);
  const tasks: GeneratorCallback[] = [];

  const init = await applicationGenerator(tree, {
    ...options,
    framework: 'none',
    e2eTestRunner: 'none',
    rootProject: true,
  });

  tasks.push(init);

  const addServerlessTask = await setupServerlessGenerator(tree, {
    ...options,
    project: options.name,
  });

  tasks.push(addServerlessTask);

  return runTasksInSerial(...tasks);
}
