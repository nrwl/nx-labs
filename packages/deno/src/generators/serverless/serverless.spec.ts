import { readJson, readProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { denoServerlessGenerator } from './serverless';

describe('serverless generator', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should setup netlify project', async () => {
    await denoServerlessGenerator(tree, {
      platform: 'netlify',
      name: 'my-netlify-app',
    });

    expect(readProjectConfiguration(tree, 'my-netlify-app')).toMatchSnapshot();
    expect(readJson(tree, 'my-netlify-app/deno.json')).toEqual({
      importMap: '../import_map.json',
    });

    expect(tree.read('my-netlify-app/netlify.toml', 'utf-8'))
      .toMatchInlineSnapshot(`
      "# Netlify Configuration File: https://docs.netlify.com/configure-builds/file-based-configuration
      [build]
        # custom directory where edge functions are located.
        # each file in this directory will be considered a separate edge function.
        edge_functions = \\"src\\"
        publish = \\"src\\"

      [functions]
        # provide all import aliases to netlify
        # https://docs.netlify.com/edge-functions/api/#import-maps
        deno_import_map = \\"../import_map.json\\"

      # Read more about declaring edge functions:
      # https://docs.netlify.com/edge-functions/declarations/#declare-edge-functions-in-netlify-toml
      [[edge_functions]]
        # this is the name of the file in the src.
        function = \\"main\\"
        # this is the route that the edge function applies to.
        path = \\"/\\"

      "
    `);
    expect(tree.exists('my-netlify-app/functions')).toBeFalsy();
    expect(tree.read('my-netlify-app/src/main.ts', 'utf-8'))
      .toMatchInlineSnapshot(`
      "/**
       * Netlify Edge Function overview:
       * https://docs.netlify.com/edge-functions/overview/
       **/

      import type { Context } from 'https://edge.netlify.com/';

      export default async function handler(req: Request, context: Context) {
        const content = \`<html>
        <body>
          <h1>Hello my-netlify-app ðŸ‘‹</h1>
        </body>
      </html>\`;

        return new Response(content, {
          status: 200,
          headers: { 'Content-Type': 'text/html;charset=utf-8' },
        });
      }
      "
    `);
  });

  it('should setup deno-deploy project', async () => {
    await denoServerlessGenerator(tree, {
      platform: 'deno-deploy',
      name: 'my-deno-deploy-app',
    });

    expect(
      readProjectConfiguration(tree, 'my-deno-deploy-app')
    ).toMatchSnapshot();
    expect(readJson(tree, 'my-deno-deploy-app/deno.json')).toEqual({
      importMap: '../import_map.json',
    });

    expect(tree.exists('my-deno-deploy-app/netlify.toml')).toBeFalsy();
    expect(tree.exists('my-deno-deploy-app/functions')).toBeFalsy();
    expect(tree.read('my-deno-deploy-app/src/main.ts', 'utf-8'))
      .toMatchInlineSnapshot(`
      "import { serve } from 'https://deno.land/std@0.181.0/http/server.ts';

      const port = Number(Deno.env.get('PORT') || 4200);

      const handler = (request: Request): Response => {
        // https://deno.com/deploy/docs/projects#environment-variables
        const region = Deno.env.get('DENO_REGION') || 'localhost';

        const body = \`<html>
        <body>
          <h1>Viewing my-deno-deploy-app from \${region} ðŸ‘‹</h1>
        </body>
      </html>\`;

        return new Response(body, {
          status: 200,
          headers: {
            'Content-Type': 'text/html;charset=utf-8',
          },
        });
      };

      console.log(\`HTTP webserver running. Access it at: http://localhost:\${port}/\`);
      await serve(handler, { port });
      "
    `);
  });

  it('should setup empty project', async () => {
    await denoServerlessGenerator(tree, {
      platform: 'none',
      name: 'my-plain-app',
    });

    expect(readProjectConfiguration(tree, 'my-plain-app')).toMatchSnapshot();
    expect(readJson(tree, 'my-plain-app/deno.json')).toEqual({
      importMap: '../import_map.json',
    });

    expect(tree.exists('my-plain-app/netlify.toml')).toBeFalsy();
    expect(tree.exists('my-plain-app/functions')).toBeFalsy();
    expect(tree.read('my-plain-app/src/main.ts', 'utf-8'))
      .toMatchInlineSnapshot(`
      "console.log('Hello my-plain-app');
      // TODO: Write awesome code ðŸ’»
      "
    `);
  });

  it('should setup in --directory', async () => {
    await denoServerlessGenerator(tree, {
      platform: 'none',
      name: 'my-plain-app',
      directory: 'nested',
    });

    expect(
      readProjectConfiguration(tree, 'nested-my-plain-app')
    ).toMatchSnapshot();
    expect(readJson(tree, 'nested/my-plain-app/deno.json')).toEqual({
      importMap: '../../import_map.json',
    });

    expect(tree.exists('nested/my-plain-app/netlify.toml')).toBeFalsy();
    expect(tree.exists('nested/my-plain-app/functions')).toBeFalsy();
    expect(tree.read('nested/my-plain-app/src/main.ts', 'utf-8'))
      .toMatchInlineSnapshot(`
      "console.log('Hello nested-my-plain-app');
      // TODO: Write awesome code ðŸ’»
      "
    `);
  });
});
