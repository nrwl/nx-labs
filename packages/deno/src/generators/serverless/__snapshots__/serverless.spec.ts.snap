// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serverless generator should setup empty project 1`] = `
Object {
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "name": "my-plain-app",
  "projectType": "application",
  "root": "my-plain-app",
  "sourceRoot": "my-plain-app/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "my-plain-app/deno.json",
        "main": "my-plain-app/src/main.ts",
        "outputFile": "dist/my-plain-app/main.js",
      },
      "outputs": Array [
        "dist/my-plain-app",
      ],
    },
    "deploy-functions": Object {
      "executor": "nx:run-commands",
      "options": Object {
        "command": "echo \\"TODO configure deploy target\\"",
      },
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "my-plain-app/deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "my-plain-app:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/my-plain-app",
        "denoConfig": "my-plain-app/deno.json",
      },
      "outputs": Array [
        "coverage/my-plain-app",
      ],
    },
  },
}
`;

exports[`serverless generator should setup in --directory 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "nested-my-plain-app",
  "projectType": "application",
  "root": "nested/my-plain-app",
  "sourceRoot": "nested/my-plain-app/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "nested/my-plain-app/deno.json",
        "main": "nested/my-plain-app/src/main.ts",
        "outputFile": "dist/nested/my-plain-app/main.js",
      },
      "outputs": Array [
        "dist/nested/my-plain-app",
      ],
    },
    "deploy-functions": Object {
      "executor": "nx:run-commands",
      "options": Object {
        "command": "echo \\"TODO configure deploy target\\"",
      },
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "nested/my-plain-app/deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "nested-my-plain-app:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/nested/my-plain-app",
        "denoConfig": "nested/my-plain-app/deno.json",
      },
      "outputs": Array [
        "coverage/nested/my-plain-app",
      ],
    },
  },
}
`;

exports[`serverless generator should setup netlify project 1`] = `
Object {
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "name": "my-netlify-app",
  "projectType": "application",
  "root": "my-netlify-app",
  "sourceRoot": "my-netlify-app/src",
  "tags": Array [],
  "targets": Object {
    "deploy-functions": Object {
      "configurations": Object {
        "production": Object {
          "command": "yarn netlify deploy --prod --site=<Your-Netlify-Site-Name>",
          "cwd": "my-netlify-app",
        },
      },
      "executor": "nx:run-commands",
      "options": Object {
        "command": "yarn netlify deploy --site=<Your-Netlify-Site-Name>",
        "cwd": "my-netlify-app",
      },
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "my-netlify-app/deno.json",
      },
    },
    "serve": Object {
      "executor": "nx:run-commands",
      "options": Object {
        "command": "yarn netlify dev",
        "cwd": "my-netlify-app",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/my-netlify-app",
        "denoConfig": "my-netlify-app/deno.json",
      },
      "outputs": Array [
        "coverage/my-netlify-app",
      ],
    },
  },
}
`;
